server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -748,4 +748,24 @@ public static XContentParser mapToXContentParser(XContentParserConfiguration con
             throw new ElasticsearchGenerationException("Failed to generate [" + source + "]", e);
         }
     }
+
+    public static Tuple<XContentParser, XContentParser> cloneParser(XContentParser parser) throws IOException {
```

Comment:
I think we should develop one way of partially cloning xcontent tree. Let's focus on doing this right in #108417

Reply:
Approach in #108417 looks good to me, i'll incorporate it once merged.